### ISSUE TITLE
Question: dealing with duplicate named objects

### ISSUE THREAD

**Issue by gha3mi** [jacobwilliams/json-fortran#548]
Hi @jacobwilliams,

I'm currently using json-fortran in the [ForOpenAI](https://github.com/gha3mi/foropenai) project. I've encountered a situation where I receive a single string in the following format that contains multiple objects with the same name, 'data':

```json
data: {
    "content": "1",
}

data: {
    "content": "2",
}

data: {
    "content": "3",
}

data: [DONE
]
```

Usually, when dealing with a single JSON object and without the object name 'data', json-fortran is easily usable:

```json
{
    "content": "1",
}
```

```Fortran
call json%initialize()
call json%deserialize(string)
call json%get("content", content, found=found)
call json%destroy()
```

However, I am uncertain whether json-fortran supports extracting content from multiple 'data' objects in this format. Could you please provide guidance or suggestions on how to handle this scenario using json-fortran?

Thank you,
Ali

**Comment by borderite** [comment]
Your data do not seem to be correctly coded. Try:
```
{
    "data": {
        "content": "1",
    },
    "data": {
        "content": "2",
    },
    "data": {
        "content": "3",
    },
    "data": ["DONE"]
}
```
json-fortran should be able to build its tree.

**Comment by gha3mi** [comment]
Thanks for your reply.

> Your data do not seem to be correctly coded.

That is an example of a response from the OpenAI API.

**Comment by borderite** [comment]
While duplicated keys are not prohibited in JSON, you still need to quote keys and put commas to separate objects. I briefly took a look at ForOpenAI's documentation, but I couldn't find a procedure that returns json-formatted result. Are you sure that ForOpenAI is supposed to give you JSON data?

**Comment by gha3mi** [comment]
There is never enough time to improve documentation;). The documentation is very basic, automatically generated using FORD.

If you have an OpenAI API key and want to test it:

1. In the `foropenai.json` file, under the `ChatCompletion` section, set `stream` to `true`. You can find this setting at: [foropenai.json Line 17](https://github.com/gha3mi/foropenai/blob/952a0eff4bb9f5bcd39bfa48e075c7901184c572/foropenai.json#L17C1-L17C24).

2. To check the response, add the following line of code `print*, response%content` after the existing code at: [foropenai_ChatCompletion.f90 Line 877](https://github.com/gha3mi/foropenai/blob/952a0eff4bb9f5bcd39bfa48e075c7901184c572/src/foropenai_ChatCompletion.f90#L877).

3. Execute `fpm run`, and then initiate a chat with gpt from the terminal, for example `"Hello. How are you?"`. Then you will be get a response in the format described [here](https://github.com/jacobwilliams/json-fortran/issues/548#issue-2068881219).

The problem is I can't modify the response from OpenAI. I have to somehow deserialize it. Now I don't even know if that response in the above format is a JSON standard or if json-fortran is unable to read it.

**Comment by borderite** [comment]
Sorry, I misunderstood your situation. I was not aware that you were the author of ForOpenAI.  While I don't have time to try your work, I made a quick experiment with chatgpt. I visited chatgpt.com and entered a question:

  _Can you list good asian grocery markets in manhattan island. Give me your answer in the JSON format._

ChatGPT's answer was

```
{
"markets": [
{
"name": "H Mart",
"location": "38 W 32nd St, New York, NY 10001",
"specialty": "Wide variety of Korean groceries, including fresh produce, meats, and specialty items."
},
{
"name": "Hong Kong Supermarket",
"location": "157 Hester St, New York, NY 10013",
"specialty": "Chinese supermarket offering a diverse range of fresh produce, seafood, and Asian groceries."
},
{
"name": "Sunrise Mart",
"location": "4 Stuyvesant St, New York, NY 10003",
"specialty": "Japanese grocery store with a selection of fresh produce, snacks, and ingredients."
},
{
"name": "M2M Asian Grocery",
"location": "55 3rd Ave, New York, NY 10003",
"specialty": "Small Japanese and Korean grocery store with a variety of snacks, sauces, and staples."
},
{
"name": "Kalustyan's",
"location": "123 Lexington Ave, New York, NY 10016",
"specialty": "Specialty store offering a wide range of international foods, including Asian groceries, spices, and ingredients."
}
]
}
```
This output is syntactically valid. 

Also, I found the following section in the chat section of the ChatGPT document. This might be of some use for you.

**response_format**
object

Optional
An object specifying the format that the model must output. Compatible with [GPT-4 Turbo](https://platform.openai.com/docs/models/gpt-4-and-gpt-4-turbo) and all GPT-3.5 Turbo models newer than gpt-3.5-turbo-1106.

_Setting to { "type": "json_object" } enables JSON mode_, which guarantees the message the model generates is valid JSON.

Important: _when using JSON mode, you must also instruct the model to produce JSON yourself via a system or user message._ Without this, the model may generate an unending stream of whitespace until the generation reaches the token limit, resulting in a long-running and seemingly "stuck" request. Also note that the message content may be partially cut off if finish_reason="length", which indicates the generation exceeded max_tokens or the conversation exceeded the max context length.

**Comment by gha3mi** [comment]
Thank you for the testing! ForOpenAI works correctly for what you tested above.

As I mentioned earlier, the only issue is what I described [here](https://github.com/jacobwilliams/json-fortran/issues/548#issuecomment-2094689034) when the stream is set to `true`. Then, the response has the format of what I described in the first [message](https://github.com/jacobwilliams/json-fortran/issues/548#issue-2068881219).

**Comment by borderite** [comment]
Your foropenapi.json has no "response_format" in the "CharCompletion" section. I am wondering if it has something to do with the problem you described. (Well, I know very little about ChatGTP. :-) )  When I added 
```"response_format": {"type": "json_object" }```
 in the "CharCompletion" section and run foropenai, it abended saying that no api key was provided, which I provided through the environment variable.
